clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java

```diff
@@ -0,0 +1,196 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl;
+
+import static io.zeebe.client.impl.BuilderUtils.appendProperty;
+import static io.zeebe.client.impl.command.ArgumentUtil.ensureNotNull;
+
+import io.grpc.ClientInterceptor;
+import io.zeebe.client.CredentialsProvider;
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.ZeebeClientBuilder;
+import io.zeebe.client.ZeebeClientCloudBuilderStep1;
+import io.zeebe.client.ZeebeClientCloudBuilderStep1.ZeebeClientCloudBuilderStep2;
+import io.zeebe.client.ZeebeClientCloudBuilderStep1.ZeebeClientCloudBuilderStep2.ZeebeClientCloudBuilderStep3;
+import io.zeebe.client.ZeebeClientCloudBuilderStep1.ZeebeClientCloudBuilderStep2.ZeebeClientCloudBuilderStep3.ZeebeClientCloudBuilderStep4;
+import io.zeebe.client.api.JsonMapper;
+import io.zeebe.client.impl.oauth.OAuthCredentialsProvider;
+import io.zeebe.client.impl.oauth.OAuthCredentialsProviderBuilder;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Properties;
+
+public class ZeebeClientCloudBuilderImpl
+    implements ZeebeClientCloudBuilderStep1,
+        ZeebeClientCloudBuilderStep2,
+        ZeebeClientCloudBuilderStep3,
+        ZeebeClientCloudBuilderStep4 {
+
+  private static final String BASE_ADDRESS = "zeebe.camunda.io";
+  private static final String BASE_AUTH_URL = "https://login.cloud.camunda.io/oauth/token";
+
+  private final ZeebeClientBuilderImpl innerBuilder = new ZeebeClientBuilderImpl();
+
+  private String clusterId;
+  private String clientId;
+  private String clientSecret;
+
+  @Override
+  public ZeebeClientCloudBuilderStep2 withClusterId(final String clusterId) {
+    this.clusterId = clusterId;
+    return this;
+  }
+
+  @Override
+  public ZeebeClientCloudBuilderStep3 withClientId(final String clientId) {
+    this.clientId = clientId;
+    return this;
+  }
+
+  @Override
+  public ZeebeClientCloudBuilderStep4 withClientSecret(final String clientSecret) {
+    this.clientSecret = clientSecret;
+    return this;
+  }
+
+  @Override
+  public ZeebeClientBuilder brokerContactPoint(final String contactPoint) {
+    return innerBuilder.gatewayAddress(contactPoint);
+  }
+
+  @Override
+  public ZeebeClientBuilder gatewayAddress(final String gatewayAddress) {
+    return innerBuilder.gatewayAddress(gatewayAddress);
+  }
+
+  @Override
+  public ZeebeClientBuilder usePlaintext() {
+    return innerBuilder.usePlaintext();
+  }
+
+  @Override
+  public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credentialsProvider) {
+    return innerBuilder.credentialsProvider(credentialsProvider);
+  }
+
+  @Override
+  public ZeebeClientBuilder withProperties(final Properties properties) {
+    return innerBuilder.withProperties(properties);
+  }
```

Comment:
I just noticed this one. I think it should allow setting the `clusterId`, `clientId`, and `clientSecret` as well via the properties file. Feel free to do this in a separate PR though.

Reply:
Do you mind if I will do it in this PR? It not so big change, but when it will be merged users could use properties already :) I will do it in a separate commit.

Reply:
Done